src/validation.py

```diff
@@ -0,0 +1,30 @@
+#  Copyright 2024 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Module for validating the webhook request."""
+
+import hashlib
+import hmac
+
+
+class SignatureValidationError(Exception):
+    """Raised when the signature validation fails."""
+
+
+def verify_signature(payload: bytes, secret_token: str, signature_header: str) -> None:
```

Comment:
As noted earlier, this probably should return a boolean for whether the signature is valid rather than raise an exception if the signature is invalid